### PR TITLE
[EFR32] Adding  add binding commands to switch

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Problem
* With the light-switch, it was not possible to bind a lighting-app to the light-switch without using the chip-tool

#### Change overview
* With the matter shell, added the feature to append a binding
* This is a first iteration with basic functionalities 

#### Testing
* Manual test with two EFR32 devices when adding a unicast and group binding

### CODE REVIEW

File: examples/light-switch-app/efr32/src/binding-handler.cpp [hunk 1]
```diff
@@ -342,6 +401,31 @@ void SwitchWorkerFunction(intptr_t context)
...
+        {
+            // Unicast connection failure can happen if peer is offline. We'll retry connection on-demand.
+            ChipLogProgress(NotSpecified,
+                            "Binding: Failed to create session for unicast binding to device " ChipLogFormatX64
+                            ": %" CHIP_ERROR_FORMAT,
+                            ChipLogValueX64(entry->nodeId), err.Format());
+        }
+    }
+
+    BindingTable::GetInstance().Add(*entry);
```
Comment (bzbarsky-apple) [inline examples/light-switch-app/efr32/src/binding-handler.cpp:423]: Shouldn't we have a single "add" function that does both this and the two notifications, called from both here and the cluster?  Having every bindings consumer have to duplicate all this logic does not make any sense.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/light-switch-app/efr32/src/binding-handler.cpp (hunk 1 above)
Comment (mkardous-silabs) [reply examples/light-switch-app/efr32/src/binding-handler.cpp]: done

File: examples/light-switch-app/efr32/src/binding-handler.cpp [hunk 2]
```diff
@@ -202,6 +205,55 @@ CHIP_ERROR ToggleSwitchCommandHandler(int argc, char ** argv)
...
+}
+
+CHIP_ERROR BindingGroupBindCommandHandler(int argc, char ** argv)
+{
+    VerifyOrReturnError(argc == 2, CHIP_ERROR_INVALID_ARGUMENT);
+
+    EmberBindingTableEntry * entry = Platform::New<EmberBindingTableEntry>();
+    entry->type                    = EMBER_MULTICAST_BINDING;
+    entry->fabricIndex             = atoi(argv[0]);
+    entry->groupId                 = atoi(argv[1]);
```
Comment (bzbarsky-apple) [inline examples/light-switch-app/efr32/src/binding-handler.cpp:235]: This needs to set `entry->local`, no?  And `entry->clusterId`, unless it's trying to bind all clusters.

File: src/app/clusters/bindings/bindings.cpp [hunk 3]
```diff
@@ -223,3 +212,21 @@ void MatterBindingPluginServerInitCallback()
 {
     registerAttributeAccessOverride(&gAttrAccess);
 }
+
+void SaveBindingEntry(EmberBindingTableEntry & entry)
```
Comment (bzbarsky-apple) [inline src/app/clusters/bindings/bindings.cpp:216]: Can this take a const ref?

File: src/app/clusters/bindings/bindings.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+#include <app/clusters/bindings/BindingManager.h>
+#include <app/util/binding-table.h>
+
+/**
+ * @brief Saves binding in storage and notifies a new binding has been added
+ *
+ * @param entry binding to save
+ */
+void SaveBindingEntry(EmberBindingTableEntry & entry);
```
Comment (bzbarsky-apple) [inline src/app/clusters/bindings/bindings.h:27]: This should really be `AddBindingEntry`, documented as adding this entry to the binding table and doing whatever persistence and notification work is involved in that.

Comment (mkardous-silabs) [reply src/app/clusters/bindings/bindings.h]: Added more specifc comments on what the function does.

File: src/app/clusters/bindings/bindings.cpp (hunk 3 above)
Comment (mkardous-silabs) [reply src/app/clusters/bindings/bindings.cpp]: yes

File: examples/light-switch-app/efr32/src/binding-handler.cpp (hunk 2 above)
Comment (mkardous-silabs) [reply examples/light-switch-app/efr32/src/binding-handler.cpp:235]: yes it should

File: src/app/clusters/bindings/bindings.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/clusters/bindings/BindingManager.h>
+#include <app/util/binding-table.h>
+
+/**
+ * @brief appends a binding to the list of bindings
+ *        This function is to be used when a device wants to add a binding to it's own table
```
Comment (bzbarsky-apple) [inline src/app/clusters/bindings/bindings.h:24]:
```suggestion
 *        This function is to be used when a device wants to add a binding to its own table
```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**